cmake/modules/FindDav1d.cmake

```diff
@@ -0,0 +1,74 @@
+#.rst:
+# FindDav1d
+# --------
+# Finds the dav1d library
+#
+# This will define the following variables::
+#
+# DAV1D_FOUND - system has dav1d
+# DAV1D_INCLUDE_DIRS - the dav1d include directory
+# DAV1D_LIBRARIES - the dav1d libraries
+#
+
+# required dav1d library version
+set(REQUIRED_DAV1D_VERSION 0.2.1)
+
+if(ENABLE_INTERNAL_DAV1D)
+  include(ExternalProject)
+
+  # Extract version
+  file(STRINGS ${CMAKE_SOURCE_DIR}/tools/depends/target/dav1d/DAV1D-VERSION VER)
+  string(REGEX MATCH "VERSION=[^ ]*$.*" DAV1D_VER "${VER}")
+  list(GET DAV1D_VER 0 DAV1D_VER)
+  string(SUBSTRING "${DAV1D_VER}" 8 -1 DAV1D_VER)
+
+  # Extract repository URL
+  string(REGEX MATCH "BASE_URL=([^ ]*)" DAV1D_BASE_URL "${VER}")
+  list(GET DAV1D_BASE_URL 0 DAV1D_BASE_URL)
+  string(SUBSTRING "${DAV1D_BASE_URL}" 9 -1 DAV1D_BASE_URL)
+
+  # allow user to override the download URL with a local tarball
+  # needed for offline build envs
+  if(DAV1D_URL)
+    get_filename_component(DAV1D_URL "${DAV1D_URL}" ABSOLUTE)
+  else()
+    set(DAV1D_URL ${DAV1D_BASE_URL}/${DAV1D_VER}/dav1d-${DAV1D_VER}.tar.gz)
+  endif()
+
+  if(VERBOSE)
+    message(STATUS "DAV1D_URL: ${DAV1D_URL}")
+  endif()
+
+  set(DAV1D_LIBRARY ${CMAKE_BINARY_DIR}/${CORE_BUILD_DIR}/lib/dav1d.a)
+  set(DAV1D_INCLUDE_DIR ${CMAKE_BINARY_DIR}/${CORE_BUILD_DIR}/include)
+
+  externalproject_add(dav1d
+                      URL ${DAV1D_URL}
+                      DOWNLOAD_NAME dav1d-${DAV1D_VER}.tar.gz
+                      DOWNLOAD_DIR ${CMAKE_BINARY_DIR}/${CORE_BUILD_DIR}/download
+                      UPDATE_COMMAND ""
+                      PATCH_COMMAND curl https://bootstrap.pypa.io/get-pip.py -o get-pip.py &&
+                                    python3 get-pip.py &&
+                                    pip install meson &&
+                                    apt-get --quiet --assume-yes install ninja-build nasm
```

Comment:
this is not needed(and wrong). We already have meson in depends: https://github.com/xbmc/xbmc/tree/master/tools/depends/native/meson
For non-depends builds (i.e. normal linux build with system libs), meson should already be installed

Reply:
thanks for the info! I assumed only dependencies listed here would be installed on linux systems:
https://github.com/xbmc/xbmc/blame/18.5-Leia/docs/README.Linux.md#L73

should I add meson and ninja to this doc?

Reply:
Edit: I think I got it. Kodi will install ninja/meson by itself. So no need to pre-install it.

Reply:
yes both should be added, as long as external dav1d(via system libs) is not available on many distros.